hepcrawl/parsers/elsevier.py

```diff
@@ -0,0 +1,555 @@
+            'prp': 'personal report',
+            'prv': 'product review',
+            'pub': 'publisher\'s Note',
+            'rem': 'removal',
+            'req': 'request for assistance',
+            'ret': 'retraction',
+            'rev': 'review',
+            'sco': 'short communication',
+            'ssu': 'short survey',
+        }
+
+
+class ElsevierParser(object):
+    """Parser for the Elsevier format.
+
+    It can be used directly by invoking the :func:`ElsevierParser.parse` method, or be
+    subclassed to customize its behavior.
+
+    Args:
+        elsevier_record (Union[str, scrapy.selector.Selector]): the record in Elsevier format to parse.
+        source (Optional[str]): if provided, sets the ``source`` everywhere in
+            the record. Otherwise, the source is extracted from the Elsevier metadata.
+    """
+    def __init__(self, elsevier_record, source=None):
+        self.root = self.get_root_node(elsevier_record)
+        if not source:
+            source = self.publisher
+        self.builder = LiteratureBuilder(source)
+
+    def parse(self):
+        """Extract a Elsevier record into an Inspire HEP record.
+
+        Returns:
+            dict: the same record in the Inspire Literature schema.
+        """
+        self.builder.add_abstract(self.abstract)
+        self.builder.add_title(self.title, subtitle=self.subtitle)
+        self.builder.add_copyright(**self.copyright)
+        self.builder.add_document_type(self.document_type)
+        self.builder.add_license(**self.license)
+        for author in self.authors:
+            self.builder.add_author(author)
+        self.builder.add_number_of_pages(self.number_of_pages)
+        self.builder.add_publication_info(**self.publication_info)
+        for collab in self.collaborations:
+            self.builder.add_collaboration(collab)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        for keyword in self.keywords:
+            self.builder.add_keyword(keyword)
+        self.builder.add_imprint_date(self.publication_date.dumps())
+        for reference in self.references:
+            self.builder.add_reference(reference)
+
+        return self.builder.record
+
+    @property
+    def references(self):
+        """Extract a Elsevier record into an Inspire HEP references record.
+
+        Returns:
+            List[dict]: an array of reference schema records, representing
+                the references in the record
+        """
+        ref_nodes = self.root.xpath('.//bib-reference')
+        return list(
+            itertools.chain.from_iterable(
+                self.get_reference(node) for node in ref_nodes
+            )
+        )
+
+
+    remove_tags_config_abstract = {
+        'allowed_tags': ['sup', 'sub'],
+        'allowed_trees': ['math'],
+        'strip': 'self::pub-id|self::issn'
+    }
+
+    @property
+    def abstract(self):
+        abstract_nodes = self.root.xpath(".//abstract-sec/simple-para")
```

Comment:
can you try to make the selectors more specific, avoiding `//` directly under the root node? this has two effects:
* you won't match wrong things by accident (e.g. metadata for the references in case it's missing from the main document).
* it's more performant, as you don't have to traverse the whole tree to check whether nodes match.

Reply:
I'll try to make them more precise, but since we use the prism namespace the document type node is not the root one, so in order to not use `//` I should use the conditional xpath containing all the top-level document type (like article, simple-article, book, book-review, simple-book etc.)